src/ast-utils.ts

```diff
@@ -109,16 +109,21 @@ export function siblingsAfter(node: ASTNode): ASTNode[] {
  * deduplicate them and prepend them in document's head.
  */
 export function stripComments(document: ASTNode) {
-  // Use of a Map keyed by comment text enables deduplication.
-  const comments: Map<string, ASTNode> = new Map();
-  dom5.nodeWalkAll(document, dom5.isCommentNode).forEach((comment: ASTNode) => {
-    comments.set(comment.data || '', comment);
-    removeElementAndNewline(comment);
-  });
-  const head = dom5.query(document, matchers.head);
-  for (const comment of comments.values()) {
-    if (isLicenseComment(comment)) {
-      prepend(head || document, comment);
+  const uniqueLicenseTexts = new Set<string>();
+  const licenseComments: ASTNode[] = [];
```

Comment:
The array contains the actual nodes though.  I use the set just to deduplicate the text in a whitespace agnostic form.